flink-runtime/src/main/java/org/apache/flink/runtime/util/EnvironmentInformation.java

```diff
@@ -41,42 +45,148 @@
+		return getVersionsInstance().gitBuildTime;
+	}
+
+	/**
+	 * @return The Instant this version of the software was built as a String using the Europe/Berlin timezone.
+	 */
+	public static String getBuildTimeString() {
+		return getVersionsInstance().gitBuildTimeStr;
+	}
+
+	/**
+	 * @return The last known commit id of this version of the software.
+	 */
+	public static String getGitCommitId() {
+		return getVersionsInstance().gitCommitId;
+	}
+
+	/**
+	 * @return The last known abbreviated commit id of this version of the software.
+	 */
+	public static String getGitCommitIdAbbrev() {
+		return getVersionsInstance().gitCommitIdAbbrev;
+	}
+
+	/**
+	 * @return The Instant of the last commit of this code.
+	 */
+	public static Instant getGitCommitTime() {
+		return getVersionsInstance().gitCommitTime;
+	}
+
+	/**
+	 * @return The Instant of the last commit of this code as a String using the Europe/Berlin timezone.
+	 */
+	public static String getGitCommitTimeString() {
+		return getVersionsInstance().gitCommitTimeStr;
 	}
 
 	/**
 	 * Returns the code revision (commit and commit date) of Flink, as generated by the Maven builds.
-	 * 
+	 *
 	 * @return The code revision.
 	 */
 	public static RevisionInformation getRevisionInformation() {
-		String revision = UNKNOWN;
-		String commitDate = UNKNOWN;
-		try (InputStream propFile = EnvironmentInformation.class.getClassLoader().getResourceAsStream(".version.properties")) {
-			if (propFile != null) {
-				Properties properties = new Properties();
-				properties.load(propFile);
-				String propRevision = properties.getProperty("git.commit.id.abbrev");
-				String propCommitDate = properties.getProperty("git.commit.time");
-				revision = propRevision != null ? propRevision : UNKNOWN;
-				commitDate = propCommitDate != null ? propCommitDate : UNKNOWN;
+		return new RevisionInformation(getGitCommitIdAbbrev(), getGitCommitTimeString());
+	}
+
+	private static final class Versions {
+		private static final Instant DEFAULT_TIME_INSTANT = Instant.EPOCH;
+		private static final String  DEFAULT_TIME_STRING  = "1970-01-01T00:00:00Z";
+
+
+		private String  projectVersion    = UNKNOWN;
+		private String  scalaVersion      = UNKNOWN;
+		private Instant gitBuildTime      = DEFAULT_TIME_INSTANT;
+		private String  gitBuildTimeStr   = DEFAULT_TIME_STRING;
+		private String  gitCommitId       = UNKNOWN;
+		private String  gitCommitIdAbbrev = UNKNOWN;
+		private Instant gitCommitTime     = DEFAULT_TIME_INSTANT;
+		private String  gitCommitTimeStr  = DEFAULT_TIME_STRING;
+
+		private static final String PROP_FILE = ".flink-runtime.version.properties";
+
+		private static final String FAIL_MESSAGE = "The file " + PROP_FILE + " has not been generated correctly. You MUST run 'mvn generate-sources' in the flink-runtime module.";
+
+		private String getProperty(Properties properties, String key) {
+			String value = properties.getProperty(key);
+			if (value == null || value.charAt(0) == '$') {
```

Comment:
The `$` branch means that it will fail if I just run it in the IDE; instead we should return the unknown value.

Reply:
Ok, I'll update that behaviour.